velox/functions/sparksql/SIMDJsonFunctions.h

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/functions/prestosql/SIMDJsonFunctions.h"
+
+using namespace simdjson;
+
+namespace facebook::velox::functions::sparksql {
+
+template <typename T>
+struct SIMDGetJsonObjectFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  // ASCII input always produces ASCII result.
+  static constexpr bool is_default_ascii_behavior = true;
+
+  FOLLY_ALWAYS_INLINE void initialize(
+      const std::vector<TypePtr>& /*inputTypes*/,
+      const core::QueryConfig& config,
+      const arg_type<Varchar>* /*json*/,
+      const arg_type<Varchar>* jsonPath) {
+    if (jsonPath != nullptr) {
+      if (jsonPath->size() > 1 && jsonPath->data()[0] == '$') {
+        jsonPath_ = removeSingleQuotes(
+            std::string_view(jsonPath->data() + 1, jsonPath->size() - 1));
+      }
+    }
+  }
+
+  FOLLY_ALWAYS_INLINE bool call(
+      out_type<Varchar>& result,
+      const arg_type<Varchar>& json,
+      const arg_type<Varchar>& jsonPath) {
+    // Spark requires the first char in jsonPath is '$'.
+    if (jsonPath.size() < 1 || jsonPath.data()[0] != '$') {
+      return false;
+    }
+    // jsonPath is "$".
+    if (jsonPath.size() == 1) {
+      result.append(json);
+      return true;
+    }
+    ParserContext ctx(json.data(), json.size());
+    ctx.parseDocument();
+    auto rawResult = jsonPath_.has_value()
+        ? ctx.jsonDoc.at_path(jsonPath_.value().data())
+        : ctx.jsonDoc.at_path(
+              removeSingleQuotes(
+                  std::string_view(jsonPath.data() + 1, jsonPath.size() - 1))
```

Comment:
> std::string_view(jsonPath.data() + 1, jsonPath.size() - 1)

this core repeats; better change the signature of removeSingleQuotes to take path as is:

```
removeSingleQuotes(path)
```